cmd/kaf/consume.go

```diff
@@ -91,15 +93,29 @@ var consumeCmd = &cobra.Command{
 				if err != nil {
 					panic(err)
 				}
+
+				// There is a race where the metadata may not have been received
+				// yet when we ask for offsets.  Retry and backoff until it
+				// (hopefully) connects.
 				offsets, err := ldr.GetAvailableOffsets(req)
+				for retries := 1; err != nil && retries <= 20; {
+					if err == sarama.ErrNotConnected {
+						time.Sleep(time.Duration(50*retries) * time.Millisecond)
+						offsets, err = ldr.GetAvailableOffsets(req)
+						retries++
+					} else {
+						break
+					}
+				}
 				if err != nil {
 					panic(err)
 				}
+
 				followOffset := offsets.GetBlock(topic, partition).Offset - 1
 
 				if follow && followOffset > 0 {
 					offset = followOffset
-					fmt.Fprintf(os.Stderr, "Starting on partition %v with offset %v\n", partition, offset)
+					outputChan <- outputMessage{error: fmt.Errorf("starting on partition %v with offset %v", partition, offset)}
```

Comment:
With this change, errors and the output message are no longer synchronized. An alternative approach here would be to append to a slice, say `om.errors = append(om.errors, fmt.Errorf("..."))` and only sending on the output channel once.

Reply:
That's a good point...  Should be an easy fix.